### PR TITLE
 increase example code of py_func, fix some wrong description of API doc

### PR DESCRIPTION
**Pull request by zhwesky2010**
### resolve #20977

fix Issue: https://github.com/PaddlePaddle/Paddle/issues/20977 

Users says he still don't know how to use this OP after reading the example code, so increase example code 2.

Fix wrong description of API doc:
- Some numpy operations (such as +/-/shape) do not support LodTenosr and must be actively converted to numpy.
image
image

### CODE REVIEW

File: doc/fluid/api_cn/layers_cn/py_func_cn.rst [hunk 1]
```diff
@@ -14,17 +14,17 @@ PaddlePaddle Fluid通过该接口在Python端注册OP。所注册的Python OP的
 此功能还可用于调试正在运行的网络，可以通过添加没有输出的 ``py_func`` 运算，并在 ``func`` 中打印输入 ``x`` 。
 
 参数:
-    - **func** （callable） - 所注册的Python OP的前向函数，运行网络时，将根据该函数与前向输入 ``x`` ，计算前向输出 ``out`` 。
-    - **x** (Variable) -  前向函数 ``func`` 的输入，可以为 Variable | tuple[Variable] | list[Variale]， 其中 Variable 为LoDTensor类型。
-    - **out** (Variable) -  前向函数 ``func`` 输出，可以为 Variable | tuple[Variable] | list[Variale]，其中 Variable 既可以为LoDTensor类型，也可以为NumPy数组。由于Paddle无法自动推断 ``out`` 的形状和数据类型，必须应事先创建 ``out`` 。
+    - **func** （callable） - 所注册的Python OP的前向函数，运行网络时，将根据该函数与前向输入 ``x`` ，计算前向输出 ``out`` ，如果在 ``func`` 中要使用Numpy函数，则必须先手动将LoDTensor转换为Numpy数组。
```
Comment (liupluswei) [inline doc/fluid/api_cn/layers_cn/py_func_cn.rst]: 最好是在上面的description中提一下，py_func的主要涉及思路就是把LoDTensor转换为numpy，然后利用numpy的API来做运算

Comment (liupluswei) [reply doc/fluid/api_cn/layers_cn/py_func_cn.rst]: 同时也附上这个链接，告诉用户怎么利用py_func来写自己的op https://www.paddlepaddle.org.cn/documentation/docs/zh/advanced_usage/development/new_op/new_python_op.html#py-func

File: doc/fluid/advanced_usage/development/new_op/new_python_op.md [hunk 2]
```diff
@@ -14,9 +14,9 @@ def py_func(func, x, out, backward_func=None, skip_vars_in_backward_input=None):
 
 其中，
 
-- `x` 是Python Op的输入变量，可以是单个 `Variable` 或者 `List[Variable]` 。
-- `out` 是Python Op的输出变量，可以是单个 `Variable` 或者 `List[Variable]` 。
-- `func` 是Python Op的前向函数。在运行网络前向时，框架会调用 `out = func(*x)` ，根据前向输入 `x` 和前向函数 `func` 计算前向输出 `out`。
+- `x` 是Python Op的输入变量，可以是单个 `Variable` 或者 `List[Variable]` 。多个Variable必须以list[Variale]的形式传入，其中Variable为LoDTensor或Tenosr。
```
Comment (liupluswei) [inline doc/fluid/advanced_usage/development/new_op/new_python_op.md]: 后续可以考虑统一为传list[Variable]

File: doc/fluid/api_cn/layers_cn/py_func_cn.rst [hunk 3]
```diff
@@ -5,7 +5,9 @@ py_func
 
 .. py:function:: paddle.fluid.layers.py_func(func, x, out, backward_func=None, skip_vars_in_backward_input=None)
 
-PaddlePaddle Fluid通过该接口在Python端注册OP。所注册的Python OP的前向函数是 ``func``, 反向函数是 ``backward_func`` 。 Paddle将在前向部分调用 ``func`` ，并在反向部分调用 ``backward_func`` （如果 ``backward_func`` 不是None)。 ``x`` 为 ``func`` 的输入，必须为LoDTensor类型； ``out``  为 ``func`` 的输出， 既可以是LoDTensor类型, 也可以是NumPy数组。
+PaddlePaddle Fluid通过py_func在Python端注册OP。py_func的设计原理在于Paddle中的LodTensor与numpy数组可以方便的互相转换，从而可使用Python中的numpy API来自定义一个Python OP。参看[如何写新的Python OP](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/advanced_usage/development/new_op/new_python_op.html)
```
Comment (liupluswei) [inline doc/fluid/api_cn/layers_cn/py_func_cn.rst]: 这里的URL会暴露在文档页面吗？

File: doc/fluid/advanced_usage/development/new_op/new_python_op.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # 如何写新的Python OP
 
-PaddlePaddle Fluid通过 `py_func` 接口支持在Python端编写op。 
+PaddlePaddle Fluid通过 `py_func` 接口支持在Python端自定义OP。 py_func的设计原理在于Paddle中的LodTensor可以与numpy数组可以方便的互相转换，从而可以使用Python中的numpy API来自定义一个Python OP。
```
Comment (liupluswei) [inline doc/fluid/advanced_usage/development/new_op/new_python_op.md:3]: 需要同步修改对应的英文文档

**Review by liupluswei — APPROVED** [review]
LGTM

**Review by Boyan-Liu — APPROVED** [review]
LGTM

**Pull request merged**